### ISSUE TITLE
[jaxrs-spec] generated folder is invalid.

### ISSUE THREAD

**Issue by HiroishiMasaya** [swagger-api/swagger-codegen#7585]
##### Description

Swagger codegen write a following line in generated pom.xml file.

  <sourceDirectory>src/main/java</sourceDirectory>

But Swagger codegen create following folder.

  src/gen/java

##### Swagger-codegen version

2.2.2
2.2.3
3.1

##### Swagger declaration file content or url

Following code is invalid.

  [source]
  swagger-codegen / modules / swagger-codegen / src / main / java / io / swagger / codegen / languages / JavaJAXRSCXFCDIServerCodegen.java 
  [code]
  sourceFolder = "src" + File.separator + "gen" + File.separator + "java";

##### Command line used for generation

swagger-codegen generate -i swagger.yml -l jaxrs-spec -o jaxrs-spec-server

**Comment by wing328** [comment]
@HiroishiMasaya yes, we noticed that as well: https://github.com/swagger-api/swagger-codegen/pull/7442#issuecomment-361043801

Would you have time to file a PR to fix the issue?

cc @jarlesat
